colly.go

```diff
@@ -158,6 +158,13 @@ type cookieJarSerializer struct {
 
 var collectorCounter uint32
 
+// The key type is unexported to prevent collisions with context keys defined in
+// other packages.
+type key int
+
+// ProxyURLKey is the context key for the request proxy address.
+const ProxyURLKey key = 0
```

Comment:
I think it would be better to stick with the `iota` here instead of creating a new type.